File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/controlApi/app/contracts/serviceStatusContract.py
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field


class ServiceStatusContract(BaseModel):
    contractName: Literal["ServiceStatus.v1"] = "ServiceStatus.v1"
    serviceName: str
    status: Literal["ONLINE", "OFFLINE", "DEGRADED"]
    checkedAt: datetime
    version: str | None = None
    details: dict[str, Any] = Field(default_factory=dict)


class PlatformHealthContract(BaseModel):
    contractName: Literal["PlatformHealth.v1"] = "PlatformHealth.v1"
    environment: str
    release: str
    overallStatus: Literal["ONLINE", "DEGRADED", "OFFLINE"]
    checkedAt: datetime
    services: list[ServiceStatusContract]

File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/controlCenter/src/contracts/serviceStatusContract.ts
export type ServiceHealthStatus = 'ONLINE' | 'OFFLINE' | 'DEGRADED'

export interface ServiceStatusContract {
  contractName: 'ServiceStatus.v1'
  serviceName: string
  status: ServiceHealthStatus
  checkedAt: string
  version?: string | null
  details?: Record<string, unknown>
}

export interface PlatformHealthContract {
  contractName: 'PlatformHealth.v1'
  environment: string
  release: string
  overallStatus: ServiceHealthStatus
  checkedAt: string
  services: ServiceStatusContract[]
}
